### PR TITLE
WebDriver: Implement `window/rect`, `window/minimize`, `window/maximize`

### PR DESCRIPTION
**Pull request by trflynn89**
[Screencast from 2022-11-02 10-13-47.webm](https://user-images.githubusercontent.com/5600524/199512413-9a632a51-8e0a-491c-a601-a081919a6e40.webm)

### CODE REVIEW

File: Userland/Services/WebDriver/Session.cpp [hunk 1]
```diff
@@ -380,6 +380,27 @@ ErrorOr<JsonValue, WebDriverError> Session::set_window_rect(JsonValue const& pay
...
+// 11.8.3 Maximize Window, https://w3c.github.io/webdriver/#dfn-maximize-window
+ErrorOr<JsonValue, WebDriverError> Session::maximize_window()
+{
+    // 1. If the remote end does not support the Maximize Window command for the current top-level browsing context for any reason, return error with error code unsupported operation.
+
+    // 2. If the current top-level browsing context is no longer open, return error with error code no such window.
+    TRY(check_for_open_top_level_browsing_context_or_return_error());
+
+    // FIXME: 3. Handle any user prompts and return its value if it is an error.
+    // FIXME: 4. Fully exit fullscreen.
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:392]: Likewise

File: Userland/Services/WebDriver/Session.cpp [hunk 2]
```diff
@@ -401,6 +401,24 @@ ErrorOr<JsonValue, WebDriverError> Session::maximize_window()
...
+// 11.8.4 Minimize Window, https://w3c.github.io/webdriver/#minimize-window
+ErrorOr<JsonValue, WebDriverError> Session::minimize_window()
+{
+    // 1. If the remote end does not support the Minimize Window command for the current top-level browsing context for any reason, return error with error code unsupported operation.
+
+    // 2. If the current top-level browsing context is no longer open, return error with error code no such window.
+    TRY(check_for_open_top_level_browsing_context_or_return_error());
+
+    // FIXME: 3. Handle any user prompts and return its value if it is an error.
+    // FIXME: 4. Fully exit fullscreen.
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:413]: Likewise

File: Userland/Services/WebDriver/Session.cpp [hunk 3]
```diff
@@ -305,6 +308,78 @@ ErrorOr<JsonValue, WebDriverError> Session::get_window_rect()
...
+    auto x = TRY(resolve_property("x"sv, x_property, NumericLimits<i32>::min(), NumericLimits<i32>::max()));
+    auto y = TRY(resolve_property("y"sv, y_property, NumericLimits<i32>::min(), NumericLimits<i32>::max()));
+
+    // 7. If the remote end does not support the Set Window Rect command for the current top-level browsing context for any reason, return error with error code unsupported operation.
+
+    // 8. If the current top-level browsing context is no longer open, return error with error code no such window.
+    TRY(check_for_open_top_level_browsing_context_or_return_error());
+
+    // FIXME: 9. Handle any user prompts and return its value if it is an error.
+    // FIXME: 10. Fully exit fullscreen.
```
[the remark this answers is not in the document]
Comment (trflynn89) [reply Userland/Services/WebDriver/Session.cpp:361]: This isn't OS-level fullscreen, it's element level (which includes the document element). We have to implement:
https://fullscreen.spec.whatwg.org/

[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Services/WebDriver/Session.cpp:361]: gotcha!

**Comment by linusg** [comment]
Ref #15551

**Review by linusg — APPROVED** [review]

File: Userland/Services/WebDriver/Client.cpp [hunk 4]
```diff
@@ -543,6 +547,46 @@ ErrorOr<JsonValue, WebDriverError> Client::handle_get_window_handles(Vector<Stri
...
+    auto* session = TRY(find_session_with_id(parameters[0]));
+    auto result = TRY(session->get_window_rect());
+    return make_json_value(result);
+}
+
+// 11.8.2 Set Window Rect, https://w3c.github.io/webdriver/#dfn-set-window-rect
+// POST /session/{session id}/window/rect
+ErrorOr<JsonValue, WebDriverError> Client::handle_set_window_rect(Vector<StringView> const& parameters, JsonValue const& payload)
+{
+    dbgln_if(WEBDRIVER_DEBUG, "Handling PSOT /session/<session_id>/window/rect");
```
[the remark this answers is not in the document]
Comment (trflynn89) [reply Userland/Services/WebDriver/Client.cpp]: :man_facepalming:

**Pull request merged**